pages/index.js

```diff
@@ -82,19 +68,25 @@ export default function HomePage() {
     setIsModalOpen(false);
   }
 
+  function handleFormSubmit(updatedTransaction) {
+    handleEditTransaction({ ...editTransaction, ...updatedTransaction });
+    closeModal();
+  }
+
   return (
     <>
       <Header />
       <main>
         <StyledTitle>Transactions</StyledTitle>
-        <EditTransactionModal
-          isModalOpen={isModalOpen}
-          onCloseModal={closeModal}
-          isEditing={isEditing}
-          transaction={updatedTransaction}
-          onHandleEditTransaction={handleEditTransaction}
-          formatNumber={formatNumber}
-        ></EditTransactionModal>
+        <Modal isModalOpen={isModalOpen} onCloseModal={closeModal}>
+          <TransactionForm
+            formHeader="Edit Transaction"
+            buttonText="Update"
+            isEditing={isEditing}
+            updatedData={editTransaction}
+            onAdd={handleFormSubmit}
```

Comment:
This naming can be better: updatedData should be `initialData`, onAdd should be `onSubmit` isEditing should be `variant` with "edit" as one option. You don't need formHeader nor ButtonText, since you can Set these internally depending on the "variant" of your Form.